src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidBrokerHelper.cs

```diff
@@ -33,22 +41,42 @@ internal class AndroidBrokerHelper
 
         private readonly Context _androidContext;
 
+        // When the broker responds, we cannot correlate back to a started task. 
+        // So we make a simplifying assumption - only one broker open session can exist at a time
+        // This semaphore is static to enforce this
+        public static SemaphoreSlim ReadyForResponse { get; set; } = new SemaphoreSlim(0);
+
         // Important: this object MUST be accessed on a background thread. Android will check this and throw otherwise.
-        private readonly AccountManager _androidAccountManager;
+        public AccountManager AndroidAccountManager { get; }
```

Comment:
Violates the separation of concerns principle - Only the AccountManagerBroker should know about this. This helper class should encapsulate all the common operations instead.

Reply:
I think you need to review this class entirely to make it more SRP (single resp principle) friendly